pkgs/by-name/li/lianad/package.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, cmake
+, copyDesktopItems
+, makeDesktopItem
+, makeWrapper
+, expat
+, fontconfig
+, freetype
+, libGL
+, systemd
+, vulkan-loader
+, xorg
+}:
+
+let
+  runtimeLibs = [
+    expat
+    fontconfig
+    freetype
+    freetype.dev
+    libGL
+    vulkan-loader
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXi
+    xorg.libXrandr
+  ];
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lianad";
+  version = "5.0"; # keep in sync with liana
+
+  src = fetchFromGitHub {
+    owner = "wizardsardine";
+    repo = "liana";
+    rev = "v${version}";
+    hash = "sha256-RkZ2HSN7IjwN3tD0UhpMeQeqkb+Y79kSWnjJZ5KPbQk=";
+  };
+
+  cargoSha256 = "sha256-6r8G4sO/aDO5VyJbmbdFz3W2b8ZmulXJZtnz8IAwcaU=";
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  buildInputs = [
+    fontconfig
+    systemd
+  ];
```

Comment:
I passed a fine comb and noticed that there was actually a lot more stuff that could be removed without affecting the build. I'm committing the bare minimum to get `lianad` + `liana-cli` to compile. That actually boils down to quite few dependencies, since all we need is a simple rust build.

Here's the diff of everything I'm removing:
```diff
diff --git a/pkgs/by-name/li/lianad/package.nix b/pkgs/by-name/li/lianad/package.nix
index 88502c9a1f61..1cb805a86639 100644
--- a/pkgs/by-name/li/lianad/package.nix
+++ b/pkgs/by-name/li/lianad/package.nix
@@ -2,34 +2,8 @@
 , stdenv
 , rustPlatform
 , fetchFromGitHub
-, pkg-config
-, cmake
-, copyDesktopItems
-, makeDesktopItem
-, makeWrapper
-, expat
-, fontconfig
-, freetype
-, libGL
-, systemd
-, vulkan-loader
-, xorg
 }:
 
-let
-  runtimeLibs = [
-    expat
-    fontconfig
-    freetype
-    freetype.dev
-    libGL
-    vulkan-loader
-    xorg.libX11
-    xorg.libXcursor
-    xorg.libXi
-    xorg.libXrandr
-  ];
-in
 rustPlatform.buildRustPackage rec {
   pname = "lianad";
   version = "5.0"; # keep in sync with liana
@@ -43,18 +17,6 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "sha256-6r8G4sO/aDO5VyJbmbdFz3W2b8ZmulXJZtnz8IAwcaU=";
 
-  nativeBuildInputs = [
-    pkg-config
-    cmake
-    copyDesktopItems
-    makeWrapper
-  ];
-
-  buildInputs = [
-    fontconfig
-    systemd
-  ];
-
   postInstall = ''
     install -Dm0644 ./contrib/lianad_config_example.toml $out/etc/liana/config.toml
   '';

```

This diff makes me wonder if some of these components could be also removed from the GUI package. For example, I don't get the point of `systemd`, since I can't identify anything that specifies an actual `systemd` service on the GUI package.

But that is probably work for a separate PR (maybe while upgrading for 6.0?).

Reply:
The GUI requires all of those (could be one or two that are not required anymore, but at the time it was the bare minimum).

I'm curious about not needing `systemd` as an input. I recall it being required for USB devices to work at all. Would you mind checking that you can connect hardware without it?

Sorry for the back and forth.

Reply:
And yes, I'll make sure to see if we can now trim down any other dependencies for GUI in the 6.0 update PR! :)

Reply:
Yeah I'll investigate a bit more but let's leave `systemd` there :)

Reply:
sounds good!

should I focus on any further changes?

Reply:
Nope, when you add systemd back I'll accept. You can also add yourself as a maintainer to `liana` as well as `lianad`.